### DISCUSSION TITLE
ConnectKit package size too large

### DISCUSSION THREAD

**Discussion by nikmace** [family/connectkit#73] [Q&A]
Hello, is there something being done about **this package being too large**? The builds become **very heavy** when importing anything from **ConnectKit**. Here is a picture of the import cost. Could there be some **optimization** of this package or its already at its smallest?

![Screenshot 2023-01-06 at 15 45 09](https://user-images.githubusercontent.com/39880364/211024501-ff49ca66-745b-46d4-bc1b-5fc267fb0abb.png)

**Comment by lochie** [comment]
@nikmace woah this doesn't seem right at all, thanks for catching this issue

I'll review when I'm back in office, might be an issue from how we recently moved our build process to a pure ES module, but will keep you notified

**Reply by nikmace** [reply]
Thanks for the reply. The package is **great** and **very useful** to us, can't really find a better alternative that works like this out of the box. It was just strange to me why would this package take up so much space. 

Just to add to this, **wagmi** is also a big boy, but I can understand why. Here is a picture of the **import cost**.

![Screenshot 2023-01-07 at 10 20 34](https://user-images.githubusercontent.com/39880364/211141227-379fb3ab-54bb-4242-81e5-1faf480b0858.png)

This is the output of our build (**nextjs**):

![Screenshot 2023-01-07 at 10 24 47](https://user-images.githubusercontent.com/39880364/211141301-cc4d5d16-2633-431d-88d2-23a9ed7cbdeb.png)
The `_app.js` chunk becomes very large. We've _used all kinds of optimizations_ but the First Load JS is not coming down.

**Reply by lochie** [reply]
Based on that output I'd say that the calculation for the import cost might actually just be incorrect as it looks to be tree shaking is working correctly (else wise `_app.js` would be way higher than **452kb**).

It's probably worth noting that the [`ethers` package](https://www.npmjs.com/package/ethers) has usually been the culprit to bundle size being high and there aren't many alternatives to this dependency yet.
